Ical.Net/DataTypes/Duration.cs

```diff
@@ -0,0 +1,222 @@
+        // As a TimeSpan always refers to exact time, we specify days as part of the hours field,
+        // because time is added as exact values rather than nominal according to RFC 5545.
+        => new Duration(hours: NullIfZero(t.Days * 24 + t.Hours), minutes: NullIfZero(t.Minutes), seconds: NullIfZero(t.Seconds));
+
+    /// <summary>
+    /// Creates an instance that represents the given time span, treating the days as nominal duration and the time part as exact.
+    /// </summary>
+    /// <remarks>
+    /// According to RFC5545 the weeks and day fields of a duration are considered nominal durations while the time fields are considered exact values.
+    /// </remarks>
+    internal static Duration FromTimeSpan(TimeSpan t)
+        => new Duration(days: NullIfZero(t.Days), hours: NullIfZero(t.Hours), minutes: NullIfZero(t.Minutes), seconds: NullIfZero(t.Seconds));
+
+    /// <summary>
+    /// Gets a value representing the time parts of the given instance.
+    /// </summary>
+    internal TimeSpan TimeAsTimeSpan
+    {
+        get
+        {
+            return new TimeSpan(
+                0,
+                Hours ?? 0,
+                Minutes ?? 0,
+                Seconds ?? 0);
+        }
+    }
+
+    /// <summary>
+    /// Gets a value representing the date parts (days and weeks) of the given instance.
+    /// </summary>
+    internal TimeSpan DateAsTimeSpan
+    {
+        get
+        {
+            return new TimeSpan(
+                (Weeks ?? 0) * 7 + (Days ?? 0),
+                0,
+                0,
+                0);
+        }
+    }
+
+    /// <summary>
+    /// Convert the instance to a <see cref="TimeSpan"/>.
+    /// </summary>
+    internal TimeSpan ToTimeSpan()
+        => new TimeSpan(
+            (Weeks ?? 0) * 7 + (Days ?? 0),
+            Hours ?? 0,
+            Minutes ?? 0,
+            Seconds ?? 0);
+
+    /// <summary>
+    /// Gets a value indicating whether the duration is zero, that is, all fields are null or 0.
+    /// </summary>
+    internal bool IsZero
+        => ((Weeks ?? 0) == 0)
+            && ((Days ?? 0) == 0)
+            && ((Hours ?? 0) == 0)
+            && ((Minutes ?? 0) == 0)
+            && ((Seconds ?? 0) == 0);
+
+    /// <summary>
+    /// Gets a value indicating whether the duration is empty, that is, all fields are null.
+    /// </summary>
+    internal bool IsEmpty
+        => (Weeks == null)
+            && (Days == null)
+            && (Hours == null)
+            && (Minutes == null)
+            && (Seconds == null);
+
+    /// <summary>
+    /// Returns a negated copy of the given instance.
+    /// </summary>
+    public static Duration operator-(Duration d) =>
+        new Duration(-d.Weeks, -d.Days, -d.Hours, -d.Minutes, -d.Seconds);
+
+    /// <inheritdoc/>
+    public override string ToString()
+        => new DurationSerializer().SerializeToString(this);
+
+    private static int? GetSign(int? v) =>
+        v switch
```

Comment:
or
```cs
v switch
{
    null => null,
    >= 0 => 1,
    _ => -1
};
```

Reply:
Nice, haven't been aware of that syntax.